### PR TITLE
Add provisioning scenario doc

### PR DESCRIPTION
**Pull request by tzumainn**
This change describes some common provisioning scenarios.

### CODE REVIEW

File: docs/provisioning-scenarios.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+openstack network create --share --provider-network-type vlan --provider-physical-network datacentre --provider-segment 1234 external-provisioning
+openstack subnet create \
+  --network external-provisioning \
+  --subnet-range 192.168.17.0/24  \
+  --ip-version 4 \
+  --gateway 192.168.17.254 \
+  --allocation-pool start=192.168.17.150,end=192.168.17.159 \
+  --dhcp external-provisioning-subnet
+openstack esi node network attach node-A external-provisioning
+openstack baremetal node power on node-A
```
Comment (larsks) [inline docs/provisioning-scenarios.md:33]: Is there a step here to let Ironic know that a node has been provisioned? I'm thinking back to our conversation with the Ironic team at the PTG.

Comment (tzumainn) [reply docs/provisioning-scenarios.md:33]: Ah - you can't actually attach a network to a node without moving it to the 'active' state, so 'esi node network attach' does that for you. So in the end, your node will be in an active state.

File: docs/provisioning-scenarios.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+## External Provisioning
+
+1. Create a Neutron network and subnet corresponding to the external provisioning network
+2. Attach the external provisioning network to the node
+3. Power on the node
+
+```
+openstack network create --share --provider-network-type vlan --provider-physical-network datacentre --provider-segment 1234 external-provisioning
+openstack subnet create \
```
Comment (naved001) [inline docs/provisioning-scenarios.md:25]: why would we need to create a subnet for an external provisioning network?

File: docs/provisioning-scenarios.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  --network external-provisioning \
+  --subnet-range 192.168.17.0/24  \
+  --ip-version 4 \
+  --gateway 192.168.17.254 \
+  --allocation-pool start=192.168.17.150,end=192.168.17.159 \
+  --dhcp external-provisioning-subnet
```
Comment (naved001) [inline docs/provisioning-scenarios.md:31]: what does this line do?

File: docs/provisioning-scenarios.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+```
+
+## External Provisioning
+
+1. Create a Neutron network and subnet corresponding to the external provisioning network
+2. Attach the external provisioning network to the node
+3. Power on the node
+
+```
+openstack network create --share --provider-network-type vlan --provider-physical-network datacentre --provider-segment 1234 external-provisioning
```
Comment (naved001) [inline docs/provisioning-scenarios.md:24]: I am fine having ironic/esi admins create networks when a specific VLAN is required, but can it be restricted for use by particular projects? A tenant would not want their provisioning network be available to other tenants.

File: docs/provisioning-scenarios.md (hunk 2 above)
Comment (tzumainn) [reply docs/provisioning-scenarios.md:25]: Ah, attaching a network to a node doesn't seem to work properly without this.

File: docs/provisioning-scenarios.md (hunk 3 above)
Comment (tzumainn) [reply docs/provisioning-scenarios.md:31]: Yeah; I guess the `--dhcp` may not be necessary.

File: docs/provisioning-scenarios.md (hunk 4 above)
Comment (tzumainn) [reply docs/provisioning-scenarios.md]: Yep - I've taken the `--share` option out.

**Comment by tzumainn** [comment]
Thanks for the review! I'll test a bit with some configuration options and update this PR with what I find.

File: docs/provisioning-scenarios.md (hunk 3 above)
Comment (tzumainn) [reply docs/provisioning-scenarios.md]: Confirmed that it is unnecessary; I've removed it!

**Comment by tzumainn** [comment]
Okay, I've tweaked the commands as per your comments, and created a separate task to add documentation about sharing networks. Let me know if there are any other changes you'd like!

File: docs/provisioning-scenarios.md (hunk 1 above)
Comment (larsks) [reply docs/provisioning-scenarios.md:33]: Interesting. So the comments that Julia had re: some sort of dummy provisioning workflow are probably not important after all?

Comment (tzumainn) [reply docs/provisioning-scenarios.md:33]: At least for this particular workflow, it does not apply.

**Review by larsks — APPROVED** [review]

**Review by naved001 — APPROVED** [review]

**Pull request merged**